### DISCUSSION TITLE
Bad address: /dev/ttyUSB0

### DISCUSSION THREAD

**Discussion by dustinmoorenet** [aqualinkd/AqualinkD#350] [Q&A] [answered]
I have a RPI4 with the USB-RS485-WE-180 serial adapter. I have aqualinkd installed as a service following the docs. The process starts up fine and the web server runs fine. But it doesn't seem to connect to my panel. It doesn't throw any errors. It just sits and waits for connection.

After trying so many things to get it to recognize the panel, I moved to try to understand the messages it is receiving from the serial adapter and finally got an error:

```
pool@pool:~ $ tree /dev/serial
/dev/serial
├── by-id
│   └── usb-FTDI_USB-RS485_Cable_FT8KYGB6-if00-port0 -> ../../ttyUSB0
└── by-path
    └── platform-3f980000.usb-usb-0:1.4:1.0-port0 -> ../../ttyUSB0

3 directories, 2 files
pool@pool:~ $ cd software/AqualinkD/release/
pool@pool:~/software/AqualinkD/release $ sudo ./serial_logger /dev/ttyUSB0 -s
AqualinkD serial_logger V2.6
Error:   RS Serial: Unable to open port: (null), error 14
Error:   Serial Log:Unable to open port: /dev/ttyUSB0
Bad address: /dev/ttyUSB0
```

If I try to run any other command like:

```
pool@pool:~/software/AqualinkD/release $ sudo ./serial_logger /dev/ttyUSB0 -d
AqualinkD serial_logger V2.6
Debug:   RS Serial: Openeded serial port /dev/ttyUSB0
Notice:  RS Serial: Port /dev/ttyUSB0 low latency mode is set
Info:    RS Serial: Port /dev/ttyUSB0 set I/O blocking attributes
Notice:  Serial Log:Logging serial information!
```

The process holds forever and I have to kill the process with `sudo kill -9 ...`. Pressing Ctrl+c doesn't halt the program.

I am starting to think something is wrong with my serial adapter. It seems genuine.

Oh, also here is the output from stty
```
pool@pool:~/software/AqualinkD/release $ sudo -s eval 'stty raw -echo < /dev/ttyUSB0; cat -vte /dev/ttyUSB0'
M-_M-y^@M-7?M-^?^[M-_M-y^@M-7M-^YM-^_^UM-_M-y^@M-^?M-_M-{M-^?M-}M-^?M-^?M-YM-_M-y^@M-7M-9M-2M-oM-_M-y^@M-^?M-_M-{M-^?M-}M-^AM-^?[M-_M-y^@M-7_M-]_yM->M-y^@M-7M-^?M-SM-CM-^?M-^?M-^?sM-_M-y^@M-7M-oM-^?M-KM-_M-y^@M-7M-^YM-^_^UM-_M-y^@M-^?M-_M-{M-^?M-}M-^?M-^?M-YM-_M-y^@M-7M-9M-2M-oM-_M-y^@M-^?M-_M-{M-^?M-}M-^AM-^?[M-_M-y^@M-7_M-]_yM->M-y^@M-7M-^?M-SM-CM-^?M-^?M-^?sM-_M-y^@M-7?M-^?^[M-_M-y^@M-7M-^YM-^_^UM-_M-y^@M-^?M-_M-{M-^?M-}M-^?M-^?M-YM-_M-y^@M-7M-9M-2M-oM-_M-y^@M-^?M-_M-{M-^?M-}M-^AM-^?[M-_M-y^@M-7M-^?M-~M-[M->M-y^@M-7_M-]_yM->M-y^@M-7M-^?M-SM-CM-^?M-^?M-^?sM-_M-y^@M-7M-mM-^?M-IM-_M-y^@M-7M-^YM-^_^UM-_M-y^@M-^?M-_M-{M-^?M-}M-^?M-^?M-YM-_M-y^@M-7M-9M-2M-oM-_M-y^@M-^?M-_M-{M-^?M-}M-^AM-^?[M-_M-y^@M-7_M-]_yM->M-y^@M-7M-^?M-SM-CM-^?M-^?M-^?sM-_M-y^@M-7?M-^?^[M-_M-y^@M-7M-^YM-^_^UM-_M-y^@M-^?M-_M-{M-^?M-}M-^?M-^?M-YM-_M-y^@M-7M-9M-2M-oM-_M-y^@M-^?M-_M-{M-^?M-}M-^
``` 

Thank you for this software and any help you can provide.

**Comment by sfeakes** [comment]
The first time you ran it, you can see that /dev/ttyUSB0 wasn't available.  So that was the problem. (USB2RS485 wasn't plugged in or had disconnected for some reason)

Second time it didn't get anything on the serial port.  Your stty output doesn't look correct, (but that could easily be a terminal issue).  I'd say you probably have a wiring issue, check data+ / - are round the right way / you have good connections.

**Reply by dustinmoorenet** [reply, marked as answer]
I had the data +/- flipped. I made a flawed assumption about which was wired to which. Funny, I disconnected and reconnected those wires so many times trying to trouble shoot and never once accidentally flipped them to the correct position.

Thank you for your help.
